dvc/daemon.py

```diff
@@ -82,6 +82,7 @@ def _spawn_posix(cmd, env):
     sys.stdin.close()
     sys.stdout.close()
     sys.stderr.close()
+    os.closerange(0, 3)
```

Comment:
Hm, this line fixes the problem? Usually the first 3 fds are stdin/stdout/stderr, are you sure this line is doing anything?

Reply:
yes, that's what surprises me. It's as if `close()` is not working, maybe they are not really closed until the interpreter shuts down? Idk.

Reply:
Or maybe stdin/stdout/stderr are not really fds 0, 1, 2 Could you check?

Reply:
The first thing that I tried to fix this was:
```python
sys.stdin = None
sys.stdout = None
sys.stderr = None
```

And it magically worked. So I thought maybe there was a garbage collection issue.
But, adding `gc.collect()` did not fix it. Then I went to `os.close()` for code clarity. :)

Reply:
> Or maybe stdin/stdout/stderr are not really fds 0, 1, 2 Could you check?

No they are 0,  1 nd 2.

Reply:
Looks like you are not first to run into this https://code.activestate.com/recipes/186101-really-closing-stdin-stdout-stderr/

Reply:
So looks like sys.* is python level and doesn't necessarily close underlying filedescriptors. I guess that explains this change.

Reply:
hmm, I searched everywhere but could not get a hint. Thank you for the link. 🙏🏼